components/style/gecko_conversions.rs

```diff
@@ -6,10 +6,50 @@
 //! Ideally, it would be in geckolib itself, but coherence
 //! forces us to keep the traits and implementations here
 
+#![allow(unsafe_code)]
+
 use app_units::Au;
+use gecko_bindings::bindings::{RawServoStyleSheetStrong, ServoComputedValuesStrong};
 use gecko_bindings::structs::nsStyleCoord_CalcValue;
+use gecko_bindings::sugar::refptr::{RefCounted, RawServoStyleSheetBorrowed, ServoComputedValuesBorrowed};
+use properties::ComputedValues;
+use std::mem::transmute;
+use std::ptr::null;
+use std::sync::Arc;
+use stylesheets::Stylesheet;
 use values::computed::{CalcLengthOrPercentage, LengthOrPercentage};
 
+macro_rules! unsafe_impl_refcounted {
+    ($strong:ident, $borrowed:ident, $servo:ident) => {
+        unsafe impl<'a> RefCounted<'a> for $servo {
+            type Strong = $strong;
+            type Borrowed = $borrowed<'a>;
+            fn into_strong(x: Arc<Self>) -> Self::Strong {
+                // we can't use a default method for this
+                // because transmute needs to know that the sizes are
+                // the same pre-monomorphization.
+                // We could use transmute_copy but that throws away
+                // the size check.
+                unsafe {transmute(x)}
+            }
+            fn null_strong() -> Self::Strong {
+                unsafe {transmute(null::<Self>())}
+            }
+            fn from_borrowed(mut x: Self::Borrowed) -> &'a Arc<Self> {
+                let arc = unsafe { &*(&mut x as *mut Self::Borrowed as *mut Arc<Self>) };
```

Comment:
Huh, yes. 

(If you've noticed, I haven't yet gotten a chance to test this, mostly because it's late)